sktime/transformers/rocket.py

```diff
@@ -0,0 +1,178 @@
+from numba import njit, prange
+import numpy as np
+import pandas as pd
+
+from .base import BaseTransformer
+
+__author__ = "Angus Dempster"
+__all__ = ["ROCKET"]
+
+class Rocket(BaseTransformer):
+    """ROCKET
+
+    RandOm Convolutional KErnel Transform
+
+    @article{dempster_etal_2019,
+      author  = {Dempster, Angus and Petitjean, Fran\c{c}ois and Webb, Geoffrey I},
+      title   = {ROCKET: Exceptionally fast and accurate time series classification using random convolutional kernels},
+      year    = {2019},
+      journal = {arXiv:1910.13051}
+    }
+
+    Parameters
+    ----------
+    input_length : int, length of input time series
+    num_kernels  : int, number of random convolutional kernels
+    num_channels : int, number of channels for multivariate time series (default 1)
+    """
+
+    def __init__(self,  input_length, num_kernels, num_channels = 1):
+        self.input_length = input_length
```

Comment:
Why do we have to specify the input length at construction? Can we not infer this from the data during fitting?

Reply:
Input length is basically just used to work out an appropriate maximum kernel dilation.  Maybe `reference_length` would be more appropriate, but would require more explanation.

It could be inferred from the input, and for time series all of the same length, in probably most cases, this would be fine.

It becomes tricker for variable-length time series.  Here, there isn't necessarily one input length or reference length which makes sense.  You could set this as the length of the longest time series, or the mean or median length, etc.  (This can be a real issue for some datasets where the length of the longest time series is orders of magnitude longer than the mean.)

I guess the training data could be an optional input to the constructor or `fit`, just to infer the dimensions.

Reply:
I like your last suggestion, I think that makes most sense: use `fit` to infer heuristics or variables that depend on the data which are then used in transform. Adding a `check_is_fitted` would be great too.